includes/class-dokan-integration.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+if ( ! class_exists( 'WPUF_Dokan_Integration' ) ) :
+
+/**
+* 
+* WPUF Dokan Integration Class
+*
+* @since  2.7
+*/
+class WPUF_Dokan_Integration{
+	
+	function __construct(){
+		
+		add_filter( 'dokan_get_dashboard_nav', array($this, 'add_wpuf_posts_page' ) );
+        add_action( 'dokan_load_custom_template', array( $this, 'load_wpuf_posts_template' ) );
+		add_filter( 'dokan_query_var_filter', array( $this, 'register_wpuf_posts_queryvar' ) );
+		add_filter( 'dokan_settings_fields', array( $this, 'dokan_wpuf_settings' ) );
+		add_filter( 'wpuf_edit_post_link', array( $this, 'generate_edit_post_link' ) );
+		
+	}
+
+	/**
+	 * Insert new URL's to the dashboard navigation bar
+	 *
+	 * @param  array  $urls 
+	 * @since  2.7
+	 * @return array  
+	 */
+	public function add_wpuf_posts_page( $urls ) {
+	    
+	    $urls['posts'] = array(
+	        'title' => __( 'Posts', 'wpuf'),
+	        'icon'  => '<i class="fa fa-wordpress"></i>',
+	        'url'   => dokan_get_navigation_url( 'posts' ),
+            'pos'   => 56
+	    );
+
+	    return $urls;
+	}
+
+	/**
+	 * Load posts template
+	 *
+	 * @param  array  $query_vars 
+	 * @since  2.7
+	 * @return void  
+	 */
+	public function load_wpuf_posts_template( $query_vars ) {
+
+    	if ( isset( $query_vars['posts'] ) ) {
+        	require_once WPUF_ROOT . '/templates/dokan/posts.php';
+    	}
+
+    }
+
+	/**
+	 * Register WPUF query var
+	 *
+	 * @param  array  $query_vars 
+	 * @since  2.7
+	 * @return void  
+	 */
+	public function register_wpuf_posts_queryvar( $query_vars ) {
+        $query_vars[] = 'posts';
+
+        return $query_vars;
+    }
+
+	/**
+	 * Dokan settings for WPUF integration
+	 *
+	 * @param  array  $settings_fields 
+	 * @since  2.7
+	 * @return array  $settings_fields  
+	 */
+    public function dokan_wpuf_settings( $settings_fields ) {
+    	$edit_page = (int) dokan_get_option( 'dashboard', 'dokan_pages' );
+
+    	$settings_fields['dokan_general']['allow_wpuf_post'] = array(
+	            'name'    => 'allow_wpuf_post',
+	            'label'   => __( 'Allow Post', 'Allow Post' ),
+	            'desc'    => __( 'Allow Vendors to submit post from dashboard area', 'wpuf' ),
+	            'type'    => 'checkbox',
+	            'default' => 'off'
+	    );
+
+	    $settings_fields['dokan_general']['wpuf_post_forms'] = array(
+            'name'    => 'wpuf_post_forms',
+            'label'   => __( 'Select Post Form', 'wpuf' ),
+            'desc'    => __( 'Select a post form that will show on the vendor dashboard.', 'wpuf' ),
+            'type'    => 'select',
+            'options' => $this->get_wpuf_post_forms(),
+            'default' => 'seller'
+        );
+
+        return $settings_fields;
+    }
+
+	/**
+	 * Get all the post forms
+	 *
+	 * @param  str  $post_type
+	 * @since  2.7
+	 * @return array  $post_forms  
+	 */
+    public function get_wpuf_post_forms( $post_type='post') {
```

Comment:
rename the function as `get_post_forms`, why use `wpuf` prefix inside the class. Don't we have any functions inside to get post forms?